server/plugin/api.go

```diff
@@ -176,14 +175,18 @@ func (p *Plugin) handleGetAllLinkedProjects(w http.ResponseWriter, r *http.Reque
 func (p *Plugin) handleUnlinkProject(w http.ResponseWriter, r *http.Request) {
 	mattermostUserID := r.Header.Get(constants.HeaderMattermostUserID)
 
-	var project *serializers.ProjectDetails
-	decoder := json.NewDecoder(r.Body)
-	if err := decoder.Decode(&project); err != nil {
+	project, err := serializers.ProjectPayloadFromJSON(r.Body)
+	if err != nil {
 		p.API.LogError(constants.ErrorDecodingBody, "Error", err.Error())
 		p.handleError(w, r, &serializers.Error{Code: http.StatusBadRequest, Message: err.Error()})
 		return
 	}
 
+	if validationErr := project.IsValid(); validationErr != "" {
```

Comment:
why return string rather than error you are returning error in every other isValid function